ksqldb-common/src/main/java/io/confluent/ksql/util/RetryUtil.java

```diff
@@ -53,6 +55,7 @@ public static void retryWithBackoff(
       final int initialWaitMs,
       final int maxWaitMs,
       final Runnable runnable,
+      final AtomicBoolean stopRetrying,
```

Comment:
nit: Rather than taking a `AtomicBoolean`, why not take a `Supplier<Boolean>`?  The latter is more general and flexible, allowing method that returns a boolean to be used.  

You can then wire this up by passing `stopRetrying::get`.

Reply:
Good suggestion.